homeassistant/components/samsungtv/config_flow.py

```diff
@@ -79,15 +78,69 @@ def _get_entry(self):
             data=data,
         )
 
+    async def _async_set_unique_id(self):
+        """Set device unique_id."""
+
+        if self._id:
+            await self.async_set_unique_id(self._id)
+            self._abort_if_unique_id_configured()
+
+        await self._async_get_and_check_device_info()
+
+        self._id = self._device_info.get("device", {}).get(ATTR_UPNP_UDN.lower())
```

Comment:
Don't set `self._id` until you know its valid

Reply:
Fixed